internal/generator/generator_util.go

```diff
@@ -201,3 +228,52 @@ func genFromSrcCode(codePath string, mapVariableName string, matchFnName string,
 
 	return nil
 }
+
+func strList2Map(strs []string) map[string]struct{} {
+	if strs == nil {
+		return nil
+	}
+
+	rv := map[string]struct{}{}
+	for _, str := range strs {
+		rv[str] = struct{}{}
+	}
+
+	return rv
+}
+
+//nolint:nilnil, mnd
+func getOverride(rawInputs []string) (map[string][]mask, error) {
+	if rawInputs == nil {
+		return nil, nil
+	}
+
+	override := map[string][]mask{}
+
+	for _, rawStr := range rawInputs {
+		// rawStr should follow the format: directive:bitmask00|bitmask01|...,bitmask10|bitmask11|...
+		subs := strings.Split(rawStr, ":")
+		if len(subs) != 2 {
+			return nil, fmt.Errorf("override parameter %s format wrong", rawStr)
+		}
+
+		directive := strings.TrimSpace(subs[0])
+		if directive == "" {
+			return nil, errors.New("directive name can't be empty")
+		}
```

Comment:
Nit: Including the offending format/line might be helpful in the case the caller has provided multiple.

```suggestion
		if directive == "" {
			return nil, fmt.Errorf("directive name is empty: override=%s", rawStr)
		}
```

Reply:
fixed: return the raw override in error